tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Cpp.cs

```diff
@@ -73,15 +77,122 @@ public class Cpp : LanguageBase
             {
                 OperationsPerSecond = opsPerSecond,
                 StandardOutput = result.StandardOutput,
-                StandardError = result.StandardError
+                StandardError = result.StandardError,
+                PackageVersions = packageVersions
```

Comment:
The naming could be improved, but `IterationResults.PackageVersions` should be populated with the **runtime** package versions.  The `packageVersions` passed into `RunAsync()` is the **requested** package versions.

So the next question for C++ is how to **verify** which package versions are being used at runtime.  In .NET this is implemented in the perf framework by inspecting the assemblies loaded at runtime.  In Java, JS, and Python this is implemented using package manager commands (e.g. `mvn depdendency:list`).

Reporting the runtime package versions is almost as important as modifying the versions themselves, since it's so easy to accidentally test the wrong versions.